pkg/services/cosmosdb/cosmos-utils.go

```diff
@@ -178,3 +182,218 @@ func deleteDatabase(
 	}
 	return nil
 }
+
+// The deployment will return success once the write region is created,
+// ignoring the status of read regions , so we must implement detection logic
+// by ourselves.
+func (c *cosmosAccountManager) waitForReadLocationsReady(
+	ctx context.Context,
+	instance service.Instance,
+) (service.InstanceDetails, error) {
+	dt := instance.Details.(*cosmosdbInstanceDetails)
+	resourceGroupName := instance.ProvisioningParameters.GetString("resourceGroup")
+	accountName := dt.DatabaseAccountName
+	databaseAccountClient := c.databaseAccountsClient
+
+	err := pollingUntilReadLocationsReady(
```

Comment:
Instead of polling this, could we instead make use of the async engine like we do waiting for parent/child to be created?